apps/cowswap-frontend/src/modules/trade/hooks/usePriorityTokenAddresses.ts

```diff
@@ -13,33 +13,62 @@ import { getUiOrderType } from 'utils/orderUtils/getUiOrderType'
 
 import { useDerivedTradeState } from './useDerivedTradeState'
 
-export function usePriorityTokenAddresses(): string[] {
+export function usePriorityTokenAddresses(): Set<string> {
   const { chainId } = useWalletInfo()
   const state = useDerivedTradeState()
 
+  const setOfTokensRef = useRef(new Set<string>())
+  const pendingRef = useRef(0)
+
   const pending = useSelector<AppState, PartialOrdersMap | undefined>((state) => {
     return state.orders?.[chainId]?.pending
   })
 
-  const pendingOrdersTokenAddresses = useMemo(() => {
-    if (!pending) return undefined
+  const inputCurrency = state?.inputCurrency
+  const outputCurrency = state?.outputCurrency
+
+  const inputCurrencyAddress = getAddress(inputCurrency)
+  const outputCurrencyAddress = getAddress(outputCurrency)
 
-    return Object.values(pending)
+  const newSetOfTokens = useMemo(() => {
+    if (!pending) return new Set<string>()
```

Comment:
To avoid redundant re-renders, it's better to create a const in the file root:
```
const EMPTY_TOKENS_SET = new Set<string>()
```

and return it here